build.rs

```diff
@@ -42,22 +42,14 @@ fn main() -> std::io::Result<()> {
     protos.sort_by(|a, b| a.cmp(b));
 
     let mut config = prost_build::Config::new();
-
-    let mut derive_string = String::from("#[derive(");
-
-    #[cfg(feature = "serde")]
-    {
-        derive_string.push_str("serde::Serialize, serde::Deserialize, ");
-    }
-
-    #[cfg(feature = "ts-gen")]
-    {
-        derive_string.push_str("specta::Type, ");
-    }
-
-    derive_string.push_str(")]");
-
-    config.type_attribute(".", derive_string.as_str());
+    config.type_attribute(
+        ".",
+        "#[cfg_attr(feature = \"serde\", derive(serde::Serialize, serde::Deserialize))]",
+    );
+    config.type_attribute(
+        ".",
+        "#[cfg_attr(feature = \"ts-gen\", derive(specta::Type))]",
+    );
 
     #[cfg(feature = "serde")]
     {
```

Comment:
Would we want to take a similar approach for the `#[serde(rename_all = \"camelCase\")]` attribute just below here as well?

Also, I think the two `#[allow(clippy::*)'`` attributes may be something that can be added independent of whether the `serde` feature is enabled or not. That's at least what my understanding was here in this PR last week: https://github.com/meshtastic/rust/pull/52#discussion_r2160170026

Reply:
That's a nice improvement, thanks @brannondorsey. `build.rs` looks way better now.